Automattic/jetpack
# AI Assistant: Introduce plugin sidebar and proofread

Related with #31946 

It brings back the changes reverted at #31979 with the proper fix.

You can check more about the `post-mortem` here: pe4Cmx-1wv-p2

## Proposed changes:

* Add new `AiAssistantPluginSidebar` component.
* Add new `Proofread` component.
* Add flag to hide sidebar.

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






Instructions to setup it:

*Jurassic Ninja*

* Spawn a new JN site.
* Download `Code Snippets` plugin.
* Introduce a new snippet to activate plugin sidebar: `define('JETPACK_AI_ASSISTANT_PLUGIN_SIDEBAR', true);`
* Goes into `Settings > Jetpack Constants`.
* Set the option `Type of blocks to load in the editor` as `Beta` and `Save changes`. 

*Locally*

* Download `Code Snippets` plugin.
* Introduce a new snippet to activate plugin sidebar: `define('JETPACK_AI_ASSISTANT_PLUGIN_SIDEBAR', true);`
* Introduce a new snippet to set Jetpack Beta blocks: `define('JETPACK_BLOCKS_VARIATION', 'beta');`

*After*:

* Go to an post.
* Check if `Jetpack` sidebar has `AI Assistant` panel.
* Click on publish.
* Check if `Pre Publish` sidebar has `AI Assistant` panel.
* Test it across P2 to check if isn't breaking other blocks.
